cudax/include/cuda/experimental/__async/stop_token.cuh

```diff
@@ -0,0 +1,488 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDAX_ASYNC_DETAIL_STOP_TOKEN_H
+#define __CUDAX_ASYNC_DETAIL_STOP_TOKEN_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__type_traits/is_nothrow_constructible.h>
+#include <cuda/std/atomic>
+#include <cuda/std/detail/libcxx/include/__threading_support>
+
+#include <cuda/experimental/__async/config.cuh>
+#include <cuda/experimental/__async/thread.cuh>
+#include <cuda/experimental/__async/utility.cuh>
+
+#if __has_include(<stop_token>) && __cpp_lib_jthread >= 201911
+#  include <stop_token>
+#endif
+
+#include <cuda/experimental/__async/prologue.cuh>
+
+// warning #20012-D: __device__ annotation is ignored on a
+// function("inplace_stop_source") that is explicitly defaulted on its first
+// declaration
+_CCCL_NV_DIAG_SUPPRESS(20012)
```

Comment:
We should not globally suppress such warnings. Is this still needed?

Reply:
if i recall, i added the __device__ annotation on the defaulted constructor to satisfy clang-cuda, but then nvcc complained so i suppressed the warning. i restore the warning to its default at the bottom of the file.